### PR TITLE
Add system loader dialog

### PR DESCRIPTION
**Pull request by m-kuhn**
fairy make apk

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2baf4da810cefb8b01212d9c3927d729092ecfd7:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-7103488e4640ec024d89c96d1053e7be2b8f8be3-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-7103488e4640ec024d89c96d1053e7be2b8f8be3-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-7103488e4640ec024d89c96d1053e7be2b8f8be3-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-7103488e4640ec024d89c96d1053e7be2b8f8be3-x86.apk)

File: src/qml/SystemLoader.qml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.12
+
+ApplicationWindow
+{
+    id: mainWindow
+    visible: true
+    width: 640
+    height: 480
```
Comment (m-kuhn) [inline src/qml/SystemLoader.qml:10]:
```suggestion
```

File: src/qml/SystemLoader.qml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+            Layout.alignment: Qt.AlignCenter
+        }
+
+        ProgressBar {
+            id: loaderProgress
+            from: 0
+            to: 100
+            value: feedback.progress
+            width: mainWindow.width
```
Comment (m-kuhn) [inline src/qml/SystemLoader.qml:33]:
```suggestion
            width: parent.width * 0.6
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b72d8cde028ab1223db165be2b3600e1bc1aaaae:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-ef2b84bb7e51c586cff1350bd97908ce56c99d63-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-ef2b84bb7e51c586cff1350bd97908ce56c99d63-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-ef2b84bb7e51c586cff1350bd97908ce56c99d63-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-ef2b84bb7e51c586cff1350bd97908ce56c99d63-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e0035ba8fcfbb780031c851aa3114dd6f921d731:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-bb6d4ef2ad329120c3a2d588709e9ef38806c227-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-bb6d4ef2ad329120c3a2d588709e9ef38806c227-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-bb6d4ef2ad329120c3a2d588709e9ef38806c227-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-bb6d4ef2ad329120c3a2d588709e9ef38806c227-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 381dc3a947a46d6e9623d49a1a9b3a8708698912:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-241fddc6045187ec0e5f28d9113fad11e8c0047a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-241fddc6045187ec0e5f28d9113fad11e8c0047a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-241fddc6045187ec0e5f28d9113fad11e8c0047a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-241fddc6045187ec0e5f28d9113fad11e8c0047a-x86.apk)

**Comment by m-kuhn** [comment]
IMO this is a requirement for 1.8 as a final release.
On older devices, the app takes currently minutes to start up. Without this, it shows a white screen. With this a loading dialog.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 703ba78439df69cb539765e7ff64a8263f95400c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-1054ff045552f69a1332653f5d82b13a096d93e6-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-1054ff045552f69a1332653f5d82b13a096d93e6-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-1054ff045552f69a1332653f5d82b13a096d93e6-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-1054ff045552f69a1332653f5d82b13a096d93e6-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 657e076dd0e2ed695ab8524296df0545e9af4abb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-8ef8bd552d52435d548ca54c3a391b6cafe4c876-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-8ef8bd552d52435d548ca54c3a391b6cafe4c876-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-8ef8bd552d52435d548ca54c3a391b6cafe4c876-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-8ef8bd552d52435d548ca54c3a391b6cafe4c876-x86.apk)

File: src/qml/SystemLoader.qml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            sourceSize.width: 200
+            sourceSize.height: 200
+
+            fillMode: Image.PreserveAspectFit
+
+            Layout.alignment: Qt.AlignCenter
+        }
+
+        Timer {
+            interval: 1000
```
Comment (m-kuhn) [inline src/qml/SystemLoader.qml:27]:
```suggestion
            interval: 3000
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for eb8efd3902ad113e96a9fecbc023099cb1f6bc09:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-b4155ad45e5a97ce28b5050dbe07269c54fd821b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-b4155ad45e5a97ce28b5050dbe07269c54fd821b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-b4155ad45e5a97ce28b5050dbe07269c54fd821b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1584-b4155ad45e5a97ce28b5050dbe07269c54fd821b-x86.apk)

**Pull request merged**